### DISCUSSION TITLE
[Closed]路由类型source能否支持匹配MAC地址

### DISCUSSION THREAD

**Discussion by silentred** [v2fly/v2ray-core#1804] [Ideas]
需求场景:
某些需要联网的智能设备不需要走proxy, 目前通过 source 指定IP列表，可以使这些设备走direct链路。但是 IP 毕竟有可能变化，如果能通过匹配MAC地址，来路由流量，就比较稳妥了。

**Comment by kulongwangzhi85** [comment]
v2ray属于应用层软件，mac属于二层数据包。拿不到mac地址。有这种需求，应该选择二层防火墙，

**Reply by universonic** [reply]
就像上一位说的，二层地址是拿不到的。你要自己在DHCP服务器上设置IP和MAC绑定关系才行。这样就不会发生变化了。

**Reply by silentred** [reply]
多谢，确实如楼上两位所说。DHCP绑定MAC和IP 比较方便。
我再提供一个思路，是否可以使用ARP定时扫描局域网，获取MAC和IP的映射。

**Reply by universonic** [reply]
你这需求太边缘太小众，不会有人做的。建议你还是自己想办法，

**Reply by simplerick-simplefun** [reply]
或者设置静态ip，绑定ip和mac。
或者自己写个小脚本，实时监控mac ip的变化，有变化之后改conf，重启v2
